qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -87,9 +87,11 @@ def __init__(
                 is provided.
             gradient: The gradient converter to be used for the probability gradients.
             quantum_instance: The quantum instance to evaluate the circuits. Note that
-                if ``sampling==True``, 'statevector_simulator' is not a valid backend for the
+                if ``sampling==True``, a statevector simulator is not a valid backend for the
                 quantum instance.
             input_gradients: Determines whether to compute gradients with respect to input data.
+                Note that this parameter is ``False`` by default, and must be explicitly set to
+                ``True`` for a proper gradient computation when using ``TorchConnector``.
```

Comment:
I have to wonder whether we could do better here for the end-user. If a user 'must' set it - can we set it on their behalf instead? Or at least check the value and fail in some way in the TorchConnector if it 'must be set'.

Reply:
I didn't add them to the base class because (correct me if I'm wrong) I thought that the instructions wouldn't show up directly when looking up i.e. "CircuitQNN". And I found it convoluted to have to refer people to the base class for something that is so "practically used". 

Adding a check makes a lot of sense! (If not changing the way it's done, I am not 100% sure of the reasons behind this parameter being set like this, but I remember that Anton changed it multiple times, maybe we should look at it further).

Reply:
@woodsp-ibm when I introduced `input_gradients`, they were set to `True` in `TorchConnector` by default. But then we discussed this with @stefan-woerner and decided that gradients with respect to input are required only in hybrid networks and thus they should be turned on explicitly. Well, we can revert this behavior and make them enabled by default again.

Reply:
Ok, as I mentioned its more to help an end user get things right - like the warnings that were added recently around output shape behavior. Logging a warning if this was detected could be helpful - though perhaps its hard to detect its use in a hybrid case specifically. We can see how things go though as it is, whether we get feedback from users etc.